1.8/administration/installing/custom/graceful-shutdown.md

```diff
@@ -0,0 +1,12 @@
+---
+post_title: Graceful Node Shutdown
+menu_order: 501
+---
+
+You can use these instructions to remove a DC/OS cluster node.
+
+1.  Add a [Mesos maintenance primitive](http://mesos.apache.org/documentation/latest/maintenance/) for the cluster node to be removed (via /machine/down). The machine will not be used for new tasks.
+1.  Use the Mesos KillPolicy ([MESOS-4909](https://issues.apache.org/jira/browse/MESOS-4909)) to specify how long, in seconds, [Marathon](https://github.com/mesosphere/marathon/commit/4a6cd7afe66c91741d835a55933da67b72c331de) will wait after sending a SIGTERM before killing the process via SIGKILL.
+1.  Write app to catch SIGHUP and SIGTERM to shutdown gracefully within the killpolicy window. 
+1.  Kill the DC/OS node, wait 21 minutes before looping (to kill another node). DC/OS is configured to rate limit automatic agent removal to 1 every 20 minutes. Do not shrink a cluster faster than this. 
```

Comment:
If you use maintenance primitives, the agent removal rate limit does not apply.  (`/mesos/master/machine/down`)

At the same time, downing a machine with maintenance primitives will also ignore kill policies, and perform force-kills immediately.